src/index.ts

```
@@ -1,33 +1,147 @@
   try {
-    const devtronPath = path.resolve('node_modules', '@electron', 'devtron');
-    const distPath = path.join(devtronPath, 'dist', 'extension');
-    const ext = await session.defaultSession.extensions.loadExtension(distPath, {
-      allowFileAccess: true,
-    });
-    console.log(`Successfully Installed extension: ${ext.name}`);
-  } catch (e) {
-    console.error('Failed to load devtron: ', e);
-    if (e instanceof Error) {
-      throw new Error(`Failed to load devtron extension: ${e.message}`);
-    } else {
-      throw new Error('Failed to load devtron extension: Unknown error');
+    const sw = await ses.serviceWorkers.startWorkerForScope(extension.url);
+    sw.startTask();
+    registerIpcListeners(ses, sw);
+  } catch (error_1) {
+    console.warn(`Failed to start Devtron service-worker (${error_1}), trying again...`);
+    /**
+     * This is a workaround for the issue where the Devtron service-worker fails to start
+     * when the Electron app is launched for the first time, or when the service worker
+     * hasn't been cached yet.
+     */
+    try {
+      const handleDetails = async (
+        event: Electron.Event,
+        details: Electron.RegistrationCompletedDetails,
+      ) => {
+        if (details.scope === extension.url) {
+          const sw = await ses.serviceWorkers.startWorkerForScope(extension.url);
+          sw.startTask();
+          registerIpcListeners(ses, sw);
+          ses.serviceWorkers.removeListener('registration-completed', handleDetails);
+          console.log(`Devtron service-worker started successfully`);
+        }
+      };
+      ses.serviceWorkers.on('registration-completed', handleDetails);
+    } catch (error_2) {
+      console.error('Failed to start Devtron service-worker:', error_2);
     }
   }
 }
 
+function install() {
+  if (isInstalled) {
+    throw new Error(
+      'Devtron has already been installed. Please avoid calling devtron.install() more than once.',
+    );
+  }
```

Wondering what everyone's thoughts are on silently returning vs. throwing an error. I think throwing might be a bit disruptive for the end user?